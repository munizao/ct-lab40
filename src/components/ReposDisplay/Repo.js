import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ repo }) => {
  return (
    <li>
      <p>{repo.name}</p>
      <p>Issues: {repo.open_issues}</p>
      <p>Forks: {repo.forks}</p>
      <p>Stars: {repo.stargazers_count}</p>
      <p>Created: {Date(repo.created_at)}</p>
      <p>Last Update: {Date(repo.updated_at)}</p>
      <p>Description: {repo.description}</p>
    </li>
  );
};

Repo.propTypes = {
  repo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    open_issues: PropTypes.number.isRequired,
    forks: PropTypes.number.isRequired,
    stargazers_count: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })
};

export default Repo;
