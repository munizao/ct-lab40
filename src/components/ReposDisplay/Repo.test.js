import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo';

describe('Repo component', () => {
  it('renders Repo', () => {
    const repo = {
      name: 'hello-world',
      open_issues: 5,
      forks: 3,
      stargazers_count: 7,
      created_at: '2020-02-17T04:30:09.735Z',
      updated_at: '2020-02-17T04:30:09.735Z',
      description: 'I just wanted to say hello to the world.'
    };
    const wrapper = shallow(<Repo repo={repo}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
