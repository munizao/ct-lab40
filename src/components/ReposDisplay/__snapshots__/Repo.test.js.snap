// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo component renders Repo 1`] = `
<li>
  <p>
    hello-world
  </p>
  <p>
    Issues: 
    5
  </p>
  <p>
    Forks: 
    3
  </p>
  <p>
    Stars: 
    7
  </p>
  <p>
    Created: 
    2/16/2020
  </p>
  <p>
    Last Update: 
    2/16/2020
  </p>
  <p>
    Description: 
    I just wanted to say hello to the world.
  </p>
</li>
`;
